test/plugin/base_test.rb

```diff
@@ -1984,54 +1994,88 @@ def test_metrics
     ].each do |config, assert_metric_value|
       [
         # Single successful request.
-        [ok_status_code, 1, 1, [1, 0, 1, 0, 0]],
+        [ok_status_code, 1, 1, [0, 0, 0]],
         # Several successful requests.
-        [ok_status_code, 2, 1, [2, 0, 2, 0, 0]],
+        [ok_status_code, 2, 1, [0, 0, 0]],
         # Single successful request with several entries.
-        [ok_status_code, 1, 2, [1, 0, 2, 0, 0]],
+        [ok_status_code, 1, 2, [0, 0, 0]],
         # Single failed request that causes logs to be dropped.
-        [client_error_status_code, 1, 1, [0, 1, 0, 1, 0]],
+        [client_error_status_code, 1, 1, [1, 1, 0]],
         # Single failed request that escalates without logs being dropped with
         # several entries.
-        [server_error_status_code, 1, 2, [0, 0, 0, 0, 2]]
+        [server_error_status_code, 1, 2, [0, 0, 2]]
       ].each do |code, request_count, entry_count, metric_values|
         clear_metrics
         setup_logging_stubs(nil, code, 'SomeMessage') do
-          (1..request_count).each do
+          (1..request_count).each do |request_index|
             d = create_driver(config)
-            (1..entry_count).each do |i|
-              d.emit('message' => log_entry(i.to_s))
+            (1..entry_count).each do |entry_index|
+              d.emit('message' => log_entry(entry_index.to_s))
             end
             # rubocop:disable Lint/HandleExceptions
             begin
               d.run
             rescue mock_error_type
             end
             # rubocop:enable Lint/HandleExceptions
+            failed_requests_count, dropped_entries_count,
+            retried_entries_count = metric_values
+
+            successful_requests_count = \
+              if code != ok_status_code
+                0
+              elsif config == ENABLE_OPENCENSUS_CONFIG
+                # The test driver instance variables can not survive between
```

Comment:
This is not ideal. There is some limitation of the test driver: e.g. it does not support a `flush` method to mock the actual behavior to bundle entries into requests as I browsed through their doc and code base.

Any suggestion to improve this is welcome. I guess potentially we can write a custom test driver to handle something like this. But I did not go all the way to do that as the additional benefit seems minimal (aka the prometheus path already covers the full story). Let me know if that is concerning.